src/components/Main/MoviesGrid.js

```diff
@@ -1,14 +1,16 @@
 import React, { useEffect } from 'react';
 import MovieCard from '../movie/MovieCard';
 import { connect } from 'react-redux';
-import { fetchMovies } from '../../actions';
-import { showMovieDetails } from '../../actions';
+import { fetchMoviesByQuery, showMovieDetails} from '../../actions';
+import { useParams } from "react-router";
 
 const MoviesGrid = (props) => {
-    const {movies, fetchMovies, showMovieDetails} = props;
+    const { movies, fetchMoviesByQuery, showMovieDetails } = props;
+    const { SearchQuery } = useParams();
```

Comment:
style suggestion: keeping in mind that we use camelCase rules for variable naming, it might be better to rename `SearchQuery` to `searchQuery` or simply `query` (same thing is applicable for `MovieId` param) 🤔

Reply:
Fixed, thank you.